src/common/tbarbase.cpp

```diff
@@ -789,24 +789,31 @@ void wxToolBarBase::UpdateWindowUI(long flags)
         if ( tool->IsSeparator() )
             continue;
 
-        int toolid = tool->GetId();
+        if ( !tool->IsControl() )
```

Comment:
Here too it's not totally obvious this doesn't change the existing behaviour...

Maybe it does it in a good way, e.g. it looks like setting text should work now whereas it didn't before, but it's not really obvious.

Reply:
My response here is basically the same as for auibar.cpp, except that here the `m_item` members are protected, rather than effectively public, so there is less opportunity for anyone to have been modifying with them.